packages/apps/src/types/plugin/plugin.ts

```diff
@@ -86,3 +82,26 @@ export interface IPlugin {
    */
   createStream?(ref: ConversationReference): IStreamer;
 }
+
+export type IPluginWithEvents<TEvents extends {}> = IBasePlugin & {
```

Comment:
same interface comment. Also is this a private type or just to be used internally byt the `App`? I'm not sure a plugin would ever want a public `emit` method since it should be emitting its events internally not allowing external acters to manually emit its events.

Reply:
also instead of having a bunch of different types for this it would be nice for it to all be in `IPlugin` as the core type for a generic plugin. From there via inheritance we expose more targeted plugin types like `ISender`.

Reply:
> same interface comment. Also is this a private type or just to be used internally byt the `App`? I'm not sure a plugin would ever want a public `emit` method since it should be emitting its events internally not allowing external acters to manually emit its events.

Responded to your earlier comment about interfaces not being possible when extending union types.

EDIT: Okay I went back and moved `__eventType` to be an optional property inside `IPlugin`. It just won't be super obvious that `__eventType` needs to be present to pick up events, it does follow convention a bit more. I also updated `ISender`